lib/core/src/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobin.hs

```diff
@@ -125,29 +153,70 @@ data SelectionCriteria = SelectionCriteria
 data SelectionResult = SelectionResult
     { inputsSelected
         :: !(NonEmpty (TxIn, TxOut))
+        -- ^ A (non-empty) list of selected inputs from the wallet's UTxO.
     , changeGenerated
         :: !(NonEmpty TokenBundle)
+        -- ^ A (non-empty) list of generated change outputs.
     , utxoRemaining
         :: !UTxOIndex
+        -- ^ UTxO remaining after performing a requested selection.
     }
     deriving (Eq, Show)
 
 -- | Represents the set of errors that may occur while performing a selection.
 --
-newtype SelectionError
+data SelectionError
     = BalanceInsufficient BalanceInsufficientError
+    | InsufficientMinCoinValues (NonEmpty InsufficientMinCoinValueError)
+    | UnableToConstructChange
+        -- ^ TODO: See if we can report how much ada is missing
+    deriving (Generic, Eq, Show)
 
 -- | Indicates that the balance of 'utxoAvailable' is insufficient to cover the
 --   balance of 'outputsToCover'.
 --
 data BalanceInsufficientError = BalanceInsufficientError
     { balanceAvailable
-        :: TokenBundle
+        :: !TokenBundle
       -- ^ The balance of 'utxoAvailable'.
     , balanceRequired
-        :: TokenBundle
+        :: !TokenBundle
       -- ^ The balance of 'outputsToCover'.
-    }
+    } deriving (Generic, Eq, Show)
+
+-- | Indicates that some of the specified outputs aren't valid and do not
+-- contain the minimum coin value expected by the protocol.
+--
+-- See also: 'prepareOutputs'.
+data InsufficientMinCoinValueError = InsufficientMinCoinValueError
+    { insufficientlyCoveredOutput
+        :: !TxOut
+        -- ^ The invalid output which doesn't hold enough coin.
```

Comment:
**Suggestion**: call this field `outputWithInsufficientAda` (or something similar)

**Reasoning**: this name more clearly relates to the fact that the ada amount is insufficient. (The output could still be **covered**, if we were to run the algorithm.)

Reply:
Fair enough

Reply:
I fixed this in ca1d961b304fe0453a1ac7c9141adc60c0e6975e.